core/keychain/keychain.go

```diff
@@ -95,7 +135,37 @@ func (kc *keychain) GetStoredKeyPairInLibP2PFormat() (crypto.PrivKey, crypto.Pub
 // It stores it in the local db and returns the key pair.
 // Warning: If there's already a key pair stored, it overrides it.
 func (kc *keychain) GenerateKeyPairWithForce() ([]byte, []byte, error) {
-	return kc.generateAndStoreKeyPair(nil)
+	return kc.generateAndStoreKeyPair(nil, "")
+}
+
+// Returns the public key currently in use in LibP2P format.
+// Returns an error if there's no public key set.
+// Unlike GetStoredKeyPairInLibP2PFormat, this method does not access the keychain
+func (kc *keychain) GetStoredPublicKey() (crypto.PubKey, error) {
```

Comment:
Should this be checking the keyring directly in addition to the store?

Reply:
Did that on purpose, that way if you have strict settings for your keychain, we don't have to be prompting for system's password whenever we try to access this public info (this was one of the optimizations so that textile client doesn't access the keychain every time it accesses a thread).